src/ui/server/shared-validation/date/index.ts

```diff
@@ -0,0 +1,126 @@
+import { endOfDay, isFuture, isValid } from 'date-fns';
+import generateValidationErrors from '../../helpers/validation';
+import createTimestampFromNumbers from '../../helpers/date/create-timestamp-from-numbers';
+import { isNumber } from '../../helpers/number';
+import { RequestBody, DateErrorMessage, ValidationErrors } from '../../../types';
+
+interface DateRulesTempType {
+  formBody: RequestBody;
+  errors: object;
+  fieldId: string;
+  errorMessages: DateErrorMessage;
+}
+
+/**
+ * dateRules
+ * Check submitted form data for errors with a date field
+ * Returns generateValidationErrors if there are any errors.
+ * @param {Express.Response.body} Express response body
+ * @param {Object} Errors object from previous validation errors
```

Comment:
do we need to add documentation fro the other variables?

Reply:
Great spot, updated